[dynamo][eval_frame] Set destroy_extra_state deleter as part of co_extra

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #107388
* #107358
* __->__ #107117
* #106917


Using the `freefunc` facility to free the ExtraState objects - https://peps.python.org/pep-0523/

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @aakhundov

anijain2305 committed:
[dynamo][eval_frame] Set destroy_extra_state deleter as part of co_extra

[ghstack-poisoned]

ezyang commented:
letting @jansel on this one

jansel commented on torch/csrc/dynamo/eval_frame.c:
```diff
@@ -420,7 +425,6 @@ inline static void set_extra_state(PyCodeObject* code, ExtraState* extra_state)
   //  the reset_code/skip, when the set_extra_state is called with
   //  NULL/SKIP_CODE.
-  destroy_extra_state(code);
   _PyCode_SetExtra((PyObject*)code, extra_index, extra_state);
```
I am assuming `_PyCode_SetExtra` calls `destroy_extra_state` if the value is already set?

(I have not confirmed this.)

anijain2305 replied:
Yes, it does

![image](https://github.com/pytorch/pytorch/assets/13822661/b41431ea-677e-492e-b55c-dd7bff534b68)

anijain2305 committed:
Update on "[dynamo][eval_frame] Set destroy_extra_state deleter as part of co_extra"



Using the `freefunc` facility to free the ExtraState objects - https://peps.python.org/pep-0523/

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78 aakhundov

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][eval_frame] Set destroy_extra_state deleter as part of co_extra"



Using the `freefunc` facility to free the ExtraState objects - https://peps.python.org/pep-0523/

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78 aakhundov

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][eval_frame] Set destroy_extra_state deleter as part of co_extra"



Using the `freefunc` facility to free the ExtraState objects - https://peps.python.org/pep-0523/

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78 aakhundov

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][eval_frame] Set destroy_extra_state deleter as part of co_extra"



Using the `freefunc` facility to free the ExtraState objects - https://peps.python.org/pep-0523/

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78 aakhundov

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][eval_frame] Set destroy_extra_state deleter as part of co_extra"



Using the `freefunc` facility to free the ExtraState objects - https://peps.python.org/pep-0523/

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78 aakhundov

[ghstack-poisoned]

anijain2305 commented:
@pytorchbot merge

anijain2305 commented:
@pytorchbot merge
